src/compiler/utilities.ts

```diff
@@ -79,6 +78,8 @@ namespace ts {
             getColumn: () => 0,
             getIndent: () => 0,
             isAtStartOfLine: () => false,
+            hasPrecedingComment: () => false,
+            hasPrecedingWhitespace: () => !!str.length && isWhiteSpaceLike(str.charCodeAt(str.length - 1)),

```

Comment:
The function is named `hasPrecedingWhitespace` but is checking for whitespace *after* the text